resources/splunk_app.rb

```diff
@@ -136,3 +148,41 @@ def config_scope
     apply_config
   end
 end
+
+class PackagedApp < SplunkApp
+  property :source_url, String, required: true
+
+  resource_name :splunk_app_package
+
+  def package_path
+    return @package_path if @package_path
+    file_cache = Pathname.new(Chef::Config['file_cache_path'])
+    @package_path = file_cache.join(CernerSplunk::PathHelpers.filename_from_url(source_url).gsub(/.spl$/, '.tgz'))
+  end
+
+  action :install do
+    return unless !version || changed?(:version)
+
+    remote_file package_path.to_s do
```

Comment:
I'm not clear on how to handle the version validation. Ensuring that the tarball version matches the expected version should eliminate any concern of an unexpected version coming through the URL. Are you suggesting ignoring this check if the installed version is a prerelease version, and also having the validation to prevent switching to a prerelease version from a release version?